crates/hyperqueue/src/dashboard/ui/screens/home/screen.rs

```diff
@@ -23,18 +25,26 @@ impl Screen for HomeScreen {
 
         self.cluster_overview.draw(layout.chart_chunk, frame);
         self.worker_util_table.draw(layout.body_chunk, frame);
+        self.worker_info_table.draw(layout.info_chunk, frame);
     }
 
     fn update(&mut self, data: &DashboardData) {
         self.worker_util_table.update(data);
+        self.worker_info_table.update(data);
         self.cluster_overview.update(data);
     }
 
     /// Handles key presses for the components of the screen
     fn handle_key(&mut self, key: Key) {
         match key {
-            Key::Down => self.worker_util_table.select_next_worker(),
-            Key::Up => self.worker_util_table.select_previous_worker(),
+            Key::Down => {
+                self.worker_util_table.select_next_worker();
+                self.worker_info_table.change_info(self.worker_util_table.get_selected_item());
```

Comment:
I would rather put the `change_info` call into `HomeScreen::update`. Or add `worker_id: Option<WorkerId>` as a parameter to `WorkerInfoTable::update`.